[WPE] WKWPE::View is refcounted

carlosgcampos opened the pull request:
#### 8104e1d037fe1ef5d6824f8d893b5df6c2691452
<pre>
[WPE] WKWPE::View is refcounted
<a href="https://bugs.webkit.org/show_bug.cgi?id=276059">https://bugs.webkit.org/show_bug.cgi?id=276059</a>

Reviewed by Michael Catanzaro.

Create methods are returning a pointer, GLib API is using
std::unique_ptr and WKTR and API tests are leaking the view.

* Source/WebKit/UIProcess/API/C/wpe/WKView.cpp:
(WKViewCreate):
(WKViewCreateDeprecated):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewCreatePage):
* Source/WebKit/UIProcess/API/wpe/WPEWebViewLegacy.h:
* Source/WebKit/UIProcess/API/wpe/WPEWebViewPlatform.h:
* Tools/TestWebKitAPI/wpe/PlatformWebViewWPE.cpp:
(TestWebKitAPI::PlatformWebView::~PlatformWebView):
* Tools/WebKitTestRunner/libwpe/PlatformWebViewClient.h:
(WTR::PlatformWebViewClient::~PlatformWebViewClient):

Canonical link: <a href="https://commits.webkit.org/280568@main">https://commits.webkit.org/280568@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0d2bf793dfb50a49fdffa5aa7d3329f716e57596

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56996 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36324 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9471 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60616 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7439 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/59124 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43947 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7629 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/46159 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/60/builds/5227 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/59026 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34111 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/49194 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/27019 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/30891 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/6525 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6444 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/52852 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/6796 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/62297 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/909 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6900 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/53416 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/912 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/49249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/53455 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/769 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8492 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/32153 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/33238 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/34323 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32984 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Tools/TestWebKitAPI/wpe/PlatformWebViewWPE.cpp:
```diff
@@ -65,6 +65,7 @@ PlatformWebView::PlatformWebView(WKPageRef relatedPage)
 PlatformWebView::~PlatformWebView()
 {
     delete m_window;
+    WKRelease(m_view);
```
Probably safer to delete the view first and window second.

carlosgcampos committed:
[WPE] WKWPE::View is refcounted
https://bugs.webkit.org/show_bug.cgi?id=276059

Reviewed by Michael Catanzaro.

Create methods are returning a pointer, GLib API is using
std::unique_ptr and WKTR and API tests are leaking the view.

* Source/WebKit/UIProcess/API/C/wpe/WKView.cpp:
(WKViewCreate):
(WKViewCreateDeprecated):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkitWebViewCreatePage):
* Source/WebKit/UIProcess/API/wpe/WPEWebViewLegacy.h:
* Source/WebKit/UIProcess/API/wpe/WPEWebViewPlatform.h:
* Tools/TestWebKitAPI/wpe/PlatformWebViewWPE.cpp:
(TestWebKitAPI::PlatformWebView::~PlatformWebView):
* Tools/WebKitTestRunner/libwpe/PlatformWebViewClient.h:
(WTR::PlatformWebViewClient::~PlatformWebViewClient):

Canonical link: https://commits.webkit.org/280568@main
